### PR TITLE
Pending trades

### PR DESCRIPTION
**Pull request by Gualdimar**
Added pending trades to the trades overview screen.
Also fixed the offer trade button state for Research agreemnt, Peace and Retract offer

### CODE REVIEW

**Comment by yairm210** [comment]
Looks good!
Pending trades should be first, since they are only relevant for one turn

File: core/src/com/unciv/ui/overviewscreen/TradesOverviewTable.kt [hunk 1]
```diff
@@ -27,10 +28,20 @@ class TradesOverviewTab(
                     else -> -1
                 }
             }
+        if (diplomacies.isNotEmpty())
```
Comment (yairm210) [inline core/src/com/unciv/ui/overviewscreen/TradesOverviewTable.kt:31]: Rename to diplomaciesWithExistingTrade

File: core/src/com/unciv/ui/overviewscreen/TradesOverviewTable.kt [hunk 2]
```diff
@@ -27,10 +28,20 @@ class TradesOverviewTab(
...
+            add("Current trades".toLabel(fontSize = Constants.headingFontSize)).padTop(10f).row()
         for (diplomacy in diplomacies) {
             for (trade in diplomacy.trades)
                 add(createTradeTable(trade, diplomacy.otherCiv())).row()
         }
+
+        diplomacies = viewingPlayer.diplomacy.values.filter { it.otherCiv().tradeRequests.any { it.requestingCiv == viewingPlayer.civName } }
```
Comment (yairm210) [inline core/src/com/unciv/ui/overviewscreen/TradesOverviewTable.kt:38]: DiplomaciesWithPendingTrade

**Comment by Gualdimar** [comment]
Updated

**Pull request merged**